File: test-suite-java/src/test/java/io/micronaut/flyway/docs/BooksFlywayConfigurationCustomizer.java
package io.micronaut.flyway.docs;

import io.micronaut.context.annotation.Requires;

// tag::imports[]
import io.micronaut.flyway.FlywayConfigurationCustomizer;
import jakarta.inject.Named;
import jakarta.inject.Singleton;
import org.flywaydb.core.api.configuration.FluentConfiguration;
import org.flywaydb.core.api.migration.JavaMigration;
// end::imports[]

@Requires(property = "spec.name", value = "BooksFlywayConfigurationCustomizerTest")
// tag::clazz[]
@Singleton
public class BooksFlywayConfigurationCustomizer implements FlywayConfigurationCustomizer { // <1>

    private static final String NAME = "books";
    private final JavaMigration[] javaMigrations;

    public BooksFlywayConfigurationCustomizer(@Named(NAME) JavaMigration[] javaMigrations) { // <2>
        this.javaMigrations = javaMigrations;
    }

    @Override
    public void customizeFluentConfiguration(FluentConfiguration fluentConfiguration) {
        if (javaMigrations != null) {
            fluentConfiguration.javaMigrations(javaMigrations); // <3>
        }
    }

    @Override
    public String getName() {
        return NAME;
    }
}
// end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/flyway/docs/BooksFlywayConfigurationCustomizer.kt
package io.micronaut.flyway.docs

import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.flyway.FlywayConfigurationCustomizer
import jakarta.inject.Named
import jakarta.inject.Singleton
import org.flywaydb.core.api.configuration.FluentConfiguration
import org.flywaydb.core.api.migration.JavaMigration
// end::imports[]

private const val NAME = "books"

@Requires(property = "spec.name", value = "BooksFlywayConfigurationCustomizerTest")
// tag::clazz[]
@Singleton
class BooksFlywayConfigurationCustomizer( // <1>
    @param:Named(NAME) private val javaMigrations: Array<JavaMigration>? // <2>
) : FlywayConfigurationCustomizer {

    override fun customizeFluentConfiguration(fluentConfiguration: FluentConfiguration) {
        if (javaMigrations != null) {
            fluentConfiguration.javaMigrations(*javaMigrations) // <3>
        }
    }

    override fun getName(): String = NAME
}
// end::clazz[]
